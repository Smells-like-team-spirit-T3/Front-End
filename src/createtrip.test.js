import { render, screen } from '@testing-library/react';
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

import { shallow } from 'enzyme';
import CreateTrip from './forms/createtrip.js';

test('Create Trip Test', () => {
  render(<CreateTrip/>);
})

describe('Create Trip input fields', () => {
  const container = shallow(<CreateTrip/>);
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  it('should have start date field', () => {
    expect(container.find('input[id="sdate"]').length).toEqual(1);
  });

  it('should have end date field', () => {
    expect(container.find('input[id="edate"]').length).toEqual(1);
  });

  it('should have destination field', () => {
    expect(container.find('input[id="dest"]').length).toEqual(1);
  });

  it('should have title field', () => {
    expect(container.find('input[id="title"]').length).toEqual(1);
  });

  it('should have participants field', () => {
    expect(container.find('input[id="participants"]').length).toEqual(1);
  });

  it('should have submit button', () => {
    expect(container.find('input[type="submit"]').length).toEqual(1);
  });
});

describe('Create Trip handleChange', () => {
    const container = shallow(<CreateTrip/>);
    it('should call handleChange', () => {
        const instance = container.instance()
        const spy = jest.spyOn(instance, 'fakeSubmit');
        console.log(instance)
        instance.forceUpdate()

        const submitButton = container.find('input[type="submit"]')
        submitButton.simulate('click')
        expect(spy).toHaveBeenCalled()
    });
});