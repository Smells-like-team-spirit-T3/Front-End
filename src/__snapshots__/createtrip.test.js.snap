// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Trip input fields should match the snapshot 1`] = `"<div class=\\"container\\"><main><h1 class=\\"title\\">Create your trip</h1><br/><form><label for=\\"sdate\\">Departure</label><input type=\\"date\\" id=\\"sdate\\" name=\\"startdate\\"/><label for=\\"edate\\">Return</label><input type=\\"date\\" id=\\"edate\\" name=\\"enddate\\"/><label for=\\"country\\">Destination</label><input type=\\"text\\" id=\\"dest\\" name=\\"dest\\" list=\\"cities\\"/><datalist id=\\"cities\\"><option value=\\"Paris\\">Paris</option><option value=\\"Berlin\\">Berlin</option><option value=\\"Prague\\">Prague</option><option value=\\"Vienna\\">Vienna</option><option value=\\"London\\">London</option><option value=\\"Basel\\">Basel</option></datalist><label for=\\"title\\">Trip Title</label><input type=\\"text\\" id=\\"title\\" name=\\"title\\"/><label for=\\"participants\\">Participants</label><input type=\\"number\\" pattern=\\"[0-9]*\\" id=\\"participants\\" name=\\"participants\\"/><input type=\\"submit\\" value=\\"Submit\\"/></form></main><footer>Powered by  Team Three</footer></div>"`;
